tools/crm_mon.c

```diff
@@ -1472,22 +1481,24 @@ main(int argc, char **argv)
     }
 
     do {
-        out->info(out,"Waiting until cluster is available on this node ...");
-
+#if CURSES_ENABLED
+        if (output_format == mon_output_console) {
+            clear();
+            refresh();
+        }
+#endif
+        out->transient(out, "Connecting to cluster...");
         rc = connect_daemons();
 
         if (rc != pcmk_rc_ok) {
-            pcmk__sleep_ms(options.reconnect_ms);
-#if CURSES_ENABLED
-            if (output_format == mon_output_console) {
-                clear();
-                refresh();
+            if ((rc == ENOTCONN) || (rc == ECONNREFUSED)) {
+                out->transient(out, "Connection failed. Retrying in %ums...",
+                               options.reconnect_ms);
             }
-#endif
-        } else if (output_format == mon_output_html && out->dest != stdout) {
-            printf("Writing html to %s ...\n", args->output_dest);
-        }
 
+            // Give some time to view all output even if we won't retry
+            pcmk__sleep_ms(options.reconnect_ms);
```

Comment:
Maybe curses' transient() should sleep like err()?

Reply:
I feel like the cons outweigh the pros, since we don't want to sleep in the common case. That would mean sleeping after "Connecting to cluster...", "Reconnecting...", etc., before we even try to do any work.

In this case, we want to sleep for `reconnect_ms` anyway before retrying if we hit `ENOTCONN` or `ECONNREFUSED`. If we somehow hit a different error (probably rare and probably indicates a bigger issue), then we give up. Putting the sleep outside the if-block just adds a one-time wait of `reconnect_ms` before we clean up and exit.